webapp/tests/Unit/Controller/API/ExecutableControllerTest.php

```diff
@@ -0,0 +1,66 @@
+<?php declare(strict_types=1);
+
+namespace App\Tests\Unit\Controller\API;
+use App\Service\DOMJudgeService;
+
+class ExecutableControllerTest extends BaseTest
+{
+    /**
+     * Test that a non logged in user can not access the executables
+     */
+    public function testExecutablesNoAnonAccess()
+    {
+        $this->verifyApiResponse('GET', "/executables/compare", 401);
+    }
+
+    /**
+     * Test that a non logged in user can not access the executables
+     */
+    public function testExecutablesNoTeamAccess()
+    {
+        $this->verifyApiResponse('GET', "/executables/compare", 403, 'demo');
+    }
+
+    /**
+     * Test that a non-existant executable can not be fetched
+     */
+    public function testExecutablesDoesNotExist()
+    {
+        $this->verifyApiResponse('GET', "/executables/volare", 404, 'admin');
+    }
+
+    /**
+     * Test that an executable can be fetched
+     */
+    public function testFetchExecutable()
+    {
+        $response = $this->verifyApiJsonResponse('GET', "/executables/compare", 200, 'admin');
+        $contents = $this->base64unzip($response);
+
+        static::assertArrayHasKey('build', $contents);
+        static::assertStringContainsString('g++ -g', $contents['build']);
+        static::assertArrayHasKey('compare.cc', $contents);
+        static::assertStringContainsString("Space!  Can't live with it, can't live without it", $contents['compare.cc']);
+    }
+
+    protected function base64unzip(string $content): array
```

Comment:
Should this be moved to Utils or Basetest?

Reply:
It can be as soon as something else wants to use it?

Reply:
It is 'testsuite' level code btw, so could be moved to BaseTest when we add tests for another endpoint that requires it. Not something to add into the main code.

Reply:
Ok makes sense, and I agree about the main code.